server/client_manager.py

```diff
@@ -377,6 +379,32 @@ def gimp_message(self, message):
                        '((status?))']
             return random.choice(message)
 
+        def check_version(self, req_release, req_major, req_minor, exact):
+            if len(self.version) < 3:
```

Comment:
wouldn't it just be easier to check if you even received a version in the first place? e.g. self.client.version is None at first and is set to the version in the "ID" message

Reply:
I'll add a check for if it just doesn't exist before it, but sanity checking the length is also important on the off-chance something goes wrong.